src/lib/apply-shim.html

```diff
@@ -133,12 +136,17 @@
     var mixinAsProperties = consumeCssProperties(valueMixin);
     var prefix = matchText.slice(0, matchText.indexOf('--'));
     var mixinValues = cssTextToMap(mixinAsProperties);
-    var oldProperties = mapGet(propertyName);
     var combinedProps = mixinValues;
-    if (oldProperties) {
+    var mixinEntry = mapGet(propertyName);
+    if (mixinEntry) {
       // NOTE: since we use mixin, the map of properties is updated here
       // and this is what we want.
-      combinedProps = Polymer.Base.mixin(oldProperties, mixinValues);
+      combinedProps = Polymer.Base.mixin(mixinEntry.properties, mixinValues);
```

Comment:
I think that can be done in the loop lower that generates `initial` values. I'll give it a shot.